browser-sign-in/OktaBrowserSignIn/TokensViewController.swift

```diff
@@ -15,89 +15,124 @@
  */
 
 import UIKit
-import OktaOidc
+import WebAuthenticationUI
 
 class TokensViewController: UIViewController {
-
-    var stateManager: OktaOidcStateManager?
+    var token: Token? {
+        didSet {
+            DispatchQueue.main.async {
+                self.showTokenInfo()
+            }
+        }
+    }
     
     @IBOutlet private var tokensView: UITextView!
-    
     @IBOutlet private var introspectButton: UIButton!
     @IBOutlet private var refreshButton: UIButton!
     @IBOutlet private var revokeButton: UIButton!
     
-    @IBOutlet private var activityIndicator: UIActivityIndicatorView!
-    
-    override func viewWillAppear(_ animated: Bool) {
-        super.viewWillAppear(animated)
-        configure()
+    override func viewDidLoad() {
+        NotificationCenter.default.addObserver(forName: .defaultCredentialChanged,
+                                               object: nil,
+                                               queue: .main) { (notification) in
+            guard let user = notification.object as? Credential else { return }
+            self.token = user.token
+        }
+        self.token = Credential.default?.token
     }
     
     @IBAction func introspectTapped() {
-        guard let accessToken = stateManager?.accessToken else { return }
-       
-        stateManager?.introspect(token: accessToken, callback: { payload, error in
+        guard let credential = Credential.default else { return  }
+        credential.introspect(.accessToken, completion: { result in
             DispatchQueue.main.async {
-                guard let isValid = payload?["active"] as? Bool else {
-                    self.showError(error?.localizedDescription ?? "Unexpected payload!")
-                    return
+                switch result {
+                case .success(let tokenInfo):
+                    guard let isValid = tokenInfo.payload["active"] as? Bool else { return }
+                    self.showAlert(title: "Access token is \(isValid ? "valid" : "invalid")!")
+                case .failure(let error):
+                    DispatchQueue.main.async {
+                        self.showError(error.localizedDescription)
+                    }
                 }
-
-                self.showAlert(title: "Access token is \(isValid ? "valid" : "invalid")!")
             }
         })
     }
     
     @IBAction func refreshTapped() {
-        stateManager?.renew(callback: { stateManager, error in
-            if let error = error {
-                self.showError(error.localizedDescription)
-                return
+        guard let credential = Credential.default
+        else {
```

Comment:
For this use case, we do not want to disable the refresh button because the idea is have a button that performs the refresh call when clicked.  I also think the usage is clear as it give a walkthrough on the failure and success path.